src/autopas/baseFunctors/CellFunctor3B.h

```diff
@@ -0,0 +1,643 @@
+      (DataLayout == DataLayoutOption::aos && cell.size() == 0)) {
+    return;
+  }
+
+  switch (DataLayout) {
+    case DataLayoutOption::aos:
+      processCellAoS<useNewton3>(cell);
+      break;
+    case DataLayoutOption::soa:
+      if (useNewton3) {
+        processCellSoAN3(cell);
+      } else {
+        processCellSoANoN3(cell);
+      }
+      break;
+  }
+}
+
+template <class Particle, class ParticleCell, class ParticleFunctor, DataLayoutOption::Value DataLayout,
+          bool useNewton3, bool bidirectional>
+void CellFunctor3B<Particle, ParticleCell, ParticleFunctor, DataLayout, useNewton3, bidirectional>::processCellPair(
+
+    ParticleCell &cell1, ParticleCell &cell2, const std::array<double, 3> &sortingDirection) {
+  if ((DataLayout == DataLayoutOption::soa &&
+       (cell1._particleSoABuffer.size() == 0 || cell2._particleSoABuffer.size() == 0)) ||
+      (DataLayout == DataLayoutOption::aos && (cell1.size() == 0 || cell2.size() == 0))) {
+    return;
+  }
+
+  switch (DataLayout) {
+    case DataLayoutOption::aos:
+      if (useNewton3) {
+        processCellPairAoSN3(cell1, cell2, sortingDirection);
+      } else {
+        processCellPairAoSNoN3(cell1, cell2, sortingDirection);
+      }
+      break;
+    case DataLayoutOption::soa:
+      if (useNewton3) {
+        processCellPairSoAN3(cell1, cell2);
+      } else {
+        processCellPairSoANoN3(cell1, cell2);
+      }
+      break;
+  }
+}
+
+template <class Particle, class ParticleCell, class ParticleFunctor, DataLayoutOption::Value DataLayout,
+          bool useNewton3, bool bidirectional>
+void CellFunctor3B<Particle, ParticleCell, ParticleFunctor, DataLayout, useNewton3, bidirectional>::processCellTriple(
+
+    ParticleCell &cell1, ParticleCell &cell2, ParticleCell &cell3, const std::array<double, 3> &sortingDirection) {
+  if ((DataLayout == DataLayoutOption::soa &&
+       (cell1._particleSoABuffer.size() == 0 || cell2._particleSoABuffer.size() == 0 ||
+        cell3._particleSoABuffer.size() == 0)) ||
+      (DataLayout == DataLayoutOption::aos && (cell1.size() == 0 || cell2.size() == 0 || cell3.size() == 0))) {
+    return;
+  }
+
+  switch (DataLayout) {
+    case DataLayoutOption::aos:
+      if (useNewton3) {
+        processCellTripleAoSN3(cell1, cell2, cell3, sortingDirection);
+      } else {
+        processCellTripleAoSNoN3(cell1, cell2, cell3, sortingDirection);
+      }
+      break;
+    case DataLayoutOption::soa:
+      if (useNewton3) {
+        processCellTripleSoAN3(cell1, cell2, cell3);
+      } else {
+        processCellTripleSoANoN3(cell1, cell2, cell3);
+      }
+      break;
+  }
+}
+
+template <class Particle, class ParticleCell, class ParticleFunctor, DataLayoutOption::Value DataLayout,
+          bool useNewton3, bool bidirectional>
+template <bool newton3>
+void CellFunctor3B<Particle, ParticleCell, ParticleFunctor, DataLayout, useNewton3, bidirectional>::processCellAoS(
```

Comment:
Have a look at the recent changes in `CellFunctor.h` that greatly reduced code duplication. Please do something similar here or even better try to reduce code duplication between the two classes